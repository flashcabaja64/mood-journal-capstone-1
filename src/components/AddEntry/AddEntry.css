.new-mood {
  width: 100%;
  display: block;
  text-align: center;
}

.new-mood-title {
  font-size: 3rem;
  margin: 40px 0 40px 0;
}

#record-mood {
  width: 100%;
  text-align: center;
}
/* Textarea box css */
div.form-section>label.label-title {
  font-size: 1.1rem;
  display: block;
  margin: 1.2rem 0 .3rem 0;
}

/* input[name="title"] {

} */

.new-mood-content {
  height: 12rem;
  width: 22rem;
  resize: none;
}
/* Radio buttons styling */
.form-section .radio-title {
  font-size: 1.1rem;
  display: block;
  margin-top: 1.2rem;
}

div.radio-form {
  margin: 0 15% 0 15%;
}

label[for="mood-type"]>span {
  font-size: 1.5rem;
}

.mood-type-explanation {
  margin: .5rem 0 2rem 0;
}

input[type="radio"] {
  border: 0px;
  width: 40px;
  height: 1.5em;
}