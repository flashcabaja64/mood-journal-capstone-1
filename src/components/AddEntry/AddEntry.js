import React, { Component } from 'react';
import EntryApiService from '../../services/entry-api-service';
import MoodContext from '../../MoodContext/MoodContext';
import TokenService from '../../services/token-service';
import ValidateError from '../ValidateError/ValidateError'
import './AddEntry.css';

export default class AddEntry extends Component {
	constructor(props) {
		super(props)

		this.state = {
			title: '',
			content: '',
			duration: '',
			mood_type: '',
			touched : {
				title: false,
				content: false,
				duration: false,
			}
		}
	}
	static contextType = MoodContext;

	handleAddMood = e => {
		e.preventDefault();
		let {title, content, duration, mood_type } = this.state
		console.log(TokenService.getUserId())
		EntryApiService.postEntry(
			TokenService.getUserId(),
			title,
			content,
			Number(duration),
			mood_type
		)
			.then(this.context.addEntry)
			.then(() => {
				title = ''
				content = ''
				mood_type = ''
			})
			.catch(this.context.setError)
			this.props.history.push('/moods')
	}

	onChangeHandle = (e) => {
		this.setState({
			[e.target.name]: e.target.value,
			touched: { [e.target.name]: true }
		})
	}

	validateTitle() {
		let name = this.state.title.trim();
		if (name.length === 0) {
			return 'Please enter more than 1 character';
		}
		if (name.length < 6) {
			return 'Please enter a name that is at least 6 characters long'
		}
	}

	validateContent() {
		let contents = this.state.content.trim();
		let words = this.state.content.split(' ')

		if (contents.length === 0) {
			return 'Please enter more than 1 character.'
		}
		if (words.length < 4){
			return 'Please enter more words.'
		}
	}

	validationDuration() {
		let num = this.state.duration;
		if(num < 0) {
			return 'Please enter a number greater than zero'
		}
	}

render() {
	let titleError = this.validateTitle();
	let contentError = this.validateContent();
	let durationError = this.validationDuration();

	return(
		<div className="new-mood">
			<header>
				<h1 className="new-mood-title">New Mood</h1>
			</header>
			<section>
				<form id="record-mood" onSubmit={this.handleAddMood}>
					<div className="form-section">
					<label htmlFor="mood-title" className="label-title">Current Mood</label>
						<input
							type="text"
							name='title'
							placeholder="Feeling weird"
							onChange={this.onChangeHandle}
							required
						/>
					<ValidateError message={titleError} />
					</div>
					<div className="form-section">
					<label htmlFor="mood-content" className="label-title">Why am I feeling this way?</label>
						<textarea
							className='new-mood-content'
							name='content'
							rows="15"
							columns="30"
							placeholder="Type your entry..."
							onChange={this.onChangeHandle}
							required
						>
						</textarea>
					<ValidateError message={contentError} />
					</div>
					<div className="form-section">
					<label htmlFor="time-mood" className="label-title">How long did the mood last?</label>
						<input
							type="number"
							id="time-mood"
							name='duration'
							placeholder="1"
							onChange={this.onChangeHandle}
							required
						/>
					<ValidateError message={durationError} />
					</div>
					<div className="radio-form">
						<div className="form-section">
						<p className="radio-title">Select mood range</p>
							<input
								type="radio"
								name='mood_type'
								value="Balanced"
								className="mood-type-radio"
								onChange={this.onChangeHandle}
								required
							/>
						<label htmlFor="mood-type">
							<span>Balanced</span>
							<div className="mood-type-explanation">This is a stable conditional mood people experience in a typical day. Things are balanced, and moods are not spiking high, and steeping low.</div>
						</label>
							<input
								type="radio"
								id="mood-type"
								name='mood_type'
								value="Low"
								className="mood-type-radio"
								onChange={this.onChangeHandle}
							/>
						<label htmlFor="mood-type">
							<span>Low</span>
							<div className="mood-type-explanation">Mood likes this has people unmotivated, and only wanting to get through the day with the least amount of effort. This could range from depressive to sad thoughts.</div>
						</label>
							<input
								type="radio"
								id="mood-type"
								name='mood_type'
								value="Extreme"
								className="mood-type-radio"
								onChange={this.onChangeHandle}
							/>
						<label htmlFor="mood-type">
							<span>Extreme</span>
							<div className="mood-type-explanation">When there is a rush of emotions that feels overpowering to where the person can still feel in control.</div>
						</label>
							<input
								type="radio"
								id="mood-type"
								name='mood_type'
								value="Erratic"
								className="mood-type-radio"
								onChange={this.onChangeHandle}
							/>
						<label htmlFor="mood-type">
							<span>Erratic</span>
							<div className="mood-type-explanation">Unable to manage the state of mood, and could, at any time, result in making regrettable and unforseen decisions.</div>
						</label>
							<input
								type="radio"
								id="mood-type"
								name='mood_type'
								value="Unsure"
								className="mood-type-radio"
								onChange={this.onChangeHandle}
							/>
						<label htmlFor="mood-type">
							<span>Unsure</span>
							<div className="mood-type-explanation">Sometimes moods cannot be labeled and we are unsure the exact mood we are experiencing. Try your best to describe it.</div>
						</label>
						</div>
					</div>
					<button
						type="submit"
						disabled={ titleError || contentError || durationError}
					>
						Submit
					</button>
					<button type="reset">Reset</button>
					</form>
				</section>
			</div>
    )
  }
}