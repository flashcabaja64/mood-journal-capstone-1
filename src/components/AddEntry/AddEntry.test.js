import React from 'react';
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom'
import ReactDOM from 'react-dom'
import toJson from 'enzyme-to-json'
import AddEntry from './AddEntry'
import renderer from 'react-test-renderer';
import enzyme from 'enzyme'

describe('AddEntry component', () => {
  
  it('renders an .AddEntry by default', () => {
    const wrapper = shallow(<AddEntry />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <AddEntry />
      </BrowserRouter>
    , div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders UI as expected', () => {
    const tree = renderer
      .create(<AddEntry />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  })

  it('should trigger onChange event', () => {
    let changeText = 'text';
    const wrapper = enzyme.shallow(<AddEntry onChange={(value) => changeText = value} />)
    wrapper.find('input[name="title"]').simulate('change', { target: {value: 'text'}});
    expect(changeText).toEqual('text')
  })
})