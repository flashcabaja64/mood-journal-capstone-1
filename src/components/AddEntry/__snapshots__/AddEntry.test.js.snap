// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEntry component renders UI as expected 1`] = `
<div
  className="new-mood"
>
  <header>
    <h1
      className="new-mood-title"
    >
      New Mood
    </h1>
  </header>
  <section>
    <form
      id="record-mood"
      onSubmit={[Function]}
    >
      <div
        className="form-section"
      >
        <label
          htmlFor="mood-title"
        >
          Current Mood
        </label>
        <input
          name="title"
          onChange={[Function]}
          placeholder="Feeling weird"
          required={true}
          type="text"
        />
        <div
          className="error-message"
        >
          Please enter more than 1 character
        </div>
      </div>
      <div
        className="form-section"
      >
        <label
          htmlFor="mood-content"
        >
          Why am I feeling this way?
        </label>
        <textarea
          columns="30"
          name="content"
          onChange={[Function]}
          placeholder="Type your entry..."
          required={true}
          rows="15"
        />
        <div
          className="error-message"
        >
          Please enter more than 1 character.
        </div>
      </div>
      <div
        className="form-section"
      >
        <label
          htmlFor="time-mood"
        >
          How long did the mood last?
        </label>
        <input
          name="duration"
          onChange={[Function]}
          placeholder="1"
          required={true}
          type="number"
        />
      </div>
      <div
        className="form-section"
      >
        <p>
          Select mood range
        </p>
        <input
          className="mood-type-radio"
          name="mood_type"
          onChange={[Function]}
          required={true}
          type="radio"
          value="Balanced"
        />
        <label
          htmlFor="mood-type"
        >
          <span>
            Balanced
          </span>
          <div
            className="mood-type-explanation"
          >
            This is a stable conditional mood people experience in a typical day. Things are balanced, and moods are not spiking high, and steeping low.
          </div>
        </label>
        <input
          className="mood-type-radio"
          name="mood_type"
          onChange={[Function]}
          type="radio"
          value="Low"
        />
        <label
          htmlFor="mood-type"
        >
          <span>
            Low
          </span>
          <div
            className="mood-type-explanation"
          >
            Mood likes this has people unmotivated, and only wanting to get through the day with the least amount of effort. This could range from depressive to sad thoughts.
          </div>
        </label>
        <input
          className="mood-type-radio"
          name="mood_type"
          onChange={[Function]}
          type="radio"
          value="Extreme"
        />
        <label
          htmlFor="mood-type"
        >
          <span>
            Extreme
          </span>
          <div
            className="mood-type-explanation"
          >
            When there is a rush of emotions that feels overpowering to where the person can still feel in control.
          </div>
        </label>
        <input
          className="mood-type-radio"
          name="mood_type"
          onChange={[Function]}
          type="radio"
          value="Erratic"
        />
        <label
          htmlFor="mood-type"
        >
          <span>
            Erratic
          </span>
          <div
            className="mood-type-explanation"
          >
            Unable to manage the state of mood, and could, at any time, result in making regrettable and unforseen decisions.
          </div>
        </label>
        <input
          className="mood-type-radio"
          name="mood_type"
          onChange={[Function]}
          type="radio"
          value="Unsure"
        />
        <label
          htmlFor="mood-type"
        >
          <span>
            Unsure
          </span>
          <div
            className="mood-type-explanation"
          >
            Sometimes moods cannot be labeled and we are unsure the exact mood we are experiencing. Try your best to describe it.
          </div>
        </label>
      </div>
      <button
        disabled="Please enter more than 1 character"
        type="submit"
      >
        Submit
      </button>
      <button
        type="reset"
      >
        Reset
      </button>
    </form>
  </section>
</div>
`;

exports[`AddEntry component renders an .AddEntry by default 1`] = `
<div
  className="new-mood"
>
  <header>
    <h1
      className="new-mood-title"
    >
      New Mood
    </h1>
  </header>
  <section>
    <form
      id="record-mood"
      onSubmit={[Function]}
    >
      <div
        className="form-section"
      >
        <label
          htmlFor="mood-title"
        >
          Current Mood
        </label>
        <input
          name="title"
          onChange={[Function]}
          placeholder="Feeling weird"
          required={true}
          type="text"
        />
        <ValidateError
          message="Please enter more than 1 character"
        />
      </div>
      <div
        className="form-section"
      >
        <label
          htmlFor="mood-content"
        >
          Why am I feeling this way?
        </label>
        <textarea
          columns="30"
          name="content"
          onChange={[Function]}
          placeholder="Type your entry..."
          required={true}
          rows="15"
        />
        <ValidateError
          message="Please enter more than 1 character."
        />
      </div>
      <div
        className="form-section"
      >
        <label
          htmlFor="time-mood"
        >
          How long did the mood last?
        </label>
        <input
          name="duration"
          onChange={[Function]}
          placeholder="1"
          required={true}
          type="number"
        />
        <ValidateError />
      </div>
      <div
        className="form-section"
      >
        <p>
          Select mood range
        </p>
        <input
          className="mood-type-radio"
          name="mood_type"
          onChange={[Function]}
          required={true}
          type="radio"
          value="Balanced"
        />
        <label
          htmlFor="mood-type"
        >
          <span>
            Balanced
          </span>
          <div
            className="mood-type-explanation"
          >
            This is a stable conditional mood people experience in a typical day. Things are balanced, and moods are not spiking high, and steeping low.
          </div>
        </label>
        <input
          className="mood-type-radio"
          name="mood_type"
          onChange={[Function]}
          type="radio"
          value="Low"
        />
        <label
          htmlFor="mood-type"
        >
          <span>
            Low
          </span>
          <div
            className="mood-type-explanation"
          >
            Mood likes this has people unmotivated, and only wanting to get through the day with the least amount of effort. This could range from depressive to sad thoughts.
          </div>
        </label>
        <input
          className="mood-type-radio"
          name="mood_type"
          onChange={[Function]}
          type="radio"
          value="Extreme"
        />
        <label
          htmlFor="mood-type"
        >
          <span>
            Extreme
          </span>
          <div
            className="mood-type-explanation"
          >
            When there is a rush of emotions that feels overpowering to where the person can still feel in control.
          </div>
        </label>
        <input
          className="mood-type-radio"
          name="mood_type"
          onChange={[Function]}
          type="radio"
          value="Erratic"
        />
        <label
          htmlFor="mood-type"
        >
          <span>
            Erratic
          </span>
          <div
            className="mood-type-explanation"
          >
            Unable to manage the state of mood, and could, at any time, result in making regrettable and unforseen decisions.
          </div>
        </label>
        <input
          className="mood-type-radio"
          name="mood_type"
          onChange={[Function]}
          type="radio"
          value="Unsure"
        />
        <label
          htmlFor="mood-type"
        >
          <span>
            Unsure
          </span>
          <div
            className="mood-type-explanation"
          >
            Sometimes moods cannot be labeled and we are unsure the exact mood we are experiencing. Try your best to describe it.
          </div>
        </label>
      </div>
      <button
        disabled="Please enter more than 1 character"
        type="submit"
      >
        Submit
      </button>
      <button
        type="reset"
      >
        Reset
      </button>
    </form>
  </section>
</div>
`;
